### PR TITLE
[TF] Top-level header fields in echo response

### PR DESCRIPTION
**Pull request by nmittler**
We currently don't have a way to distinguish between request and response headers when checking echo results. This promotes request and response headers to a top-level field in echo.Response and provides tools for validating headers.

**Please provide a description of this PR:**

### CODE REVIEW

File: tests/integration/pilot/common/routing.go [hunk 1]
```diff
@@ -520,10 +524,12 @@ spec:
 							Count:    1,
 							Check: check.And(
 								check.OK(),
-								check.WithInfo("CORS preflight header", check.Key("Access-Control-Allow-Origin", "cors.com")),
-								check.WithInfo("CORS preflight header", check.Key("Access-Control-Allow-Methods", "POST,GET")),
-								check.WithInfo("CORS preflight header", check.Key("Access-Control-Allow-Headers", "X-Foo-Bar,X-Foo-Baz")),
-								check.WithInfo("CORS preflight header", check.Key("Access-Control-Max-Age", "86400"))),
+								check.ResponseHeaders(map[string]string{
```
Comment (howardjohn) [inline tests/integration/pilot/common/routing.go:527]: nit: slightly prefer taking in 2 strings as input. This is likely one of the only cases we will want to check multiple headers and can just call the Response Header checker multiple times

File: tests/integration/telemetry/outboundtrafficpolicy/helper.go [hunk 2]
```diff
@@ -259,29 +256,10 @@ func RunExternalRequest(cases []*TestCase, prometheus prometheus.Instance, mode
...
-								for k, v := range tc.Expected.Metadata {
-									if got := r.RawResponse[k]; got != v {
-										return fmt.Errorf("expected metadata %v=%v, got %q", k, v, got)
-									}
-								}
-							}
-							return nil
-						},
+						Check: check.And(
+							check.NoError(),
```
Comment (howardjohn) [inline tests/integration/telemetry/outboundtrafficpolicy/helper.go:260]: I think we expect error in some cases

File: tests/integration/pilot/common/routing.go (hunk 1 above)
Comment (howardjohn) [reply tests/integration/pilot/common/routing.go:527]: Ok actually there was one other usage it looks like. Still my preference but I'm fine either way

**Review by howardjohn — COMMENTED** [review]
did not approve due to the telemetry test thing, not the map vs string input - let me know if I just misunderstood

File: tests/integration/pilot/common/routing.go (hunk 1 above)
Comment (nmittler) [reply tests/integration/pilot/common/routing.go:523]: I've added both and one calls the other.

**Review by bianpengyuan — APPROVED** [review]

**Pull request merged**